src/pages/studyView/StudyViewPage.tsx

```diff
@@ -5,70 +5,135 @@ import $ from 'jquery';
 
-        invoke:()=>{
+    public getChartFilters( attributeId: string,
+                            clinicalDataType: "SAMPLE" | "PATIENT"): string[] {
+        let id = [clinicalDataType,attributeId].join('_')
+        return this.propFilters[id] ? this.propFilters[id].values : []
+    }
 
-            return getMutatedGenes(this.prelimParam);
 
+    readonly clinicalAttributes = remoteData({
+        invoke: () => {
+            return defaultClient.getAllClinicalAttributesInStudyUsingGET({
+                'studyId': this.studyId
+            }).then((attributes:ClinicalAttribute[])=>{
+                let attributeIds = [...this.sampleAttrIds,...this.patientAttrIds];
+                return attributes.filter(attribute => {
+                    if(_.includes(attributeIds, attribute.clinicalAttributeId)){
+                        return true;
+                    }
+                    return false;
+                });
+			});
         }
-
     });
 
+    readonly sampleAttributesData = remoteData<ClinicalAttributeData>({
+        await: ()=>[this.clinicalAttributes],
+        invoke: () => {
+            return internalClient.fetchClinicalDataCountsUsingPOST({
+                'studyId': this.studyId,
+                'clinicalDataType': "SAMPLE",
+                'clinicalDataCountFilter': {
+                    'attributeIds': this.sampleAttrIds,
+                    'filter': this.filter
+                }
+            })
+        },
+		default: {}
+    });
 
-    readonly mutatedGenes = remoteData({
-
-        await:()=>[
-
-        ],
-
-        invoke:()=>{
-
-            return getMutatedGenes(this.filterTerm);
-
-        }
+    readonly patientAttributesData = remoteData<ClinicalAttributeData>({
+        await: ()=>[this.clinicalAttributes],
+        invoke: () => {
+            return internalClient.fetchClinicalDataCountsUsingPOST({
+                'studyId': this.studyId,
+                'clinicalDataType': "PATIENT",
+                'clinicalDataCountFilter': {
+                    'attributeIds': this.patientAttrIds,
+                    'filter': this.filter
+                }
+            })
+        },
+		default: {}
+    });
 
+    readonly cinicalAttributeData = remoteData<{[attrId:string]:ClinicalAttributeDataWithMeta}>({
+        await: ()=>[this.sampleAttributesData,this.patientAttributesData],
+        invoke: async ()=>{
+
+            let sampleAttrData = this.sampleAttributesData.result
+            let patientAttrData = this.patientAttributesData.result
+            let result:{[attrId:string]:ClinicalAttributeDataWithMeta} = {}
+            Object.keys(sampleAttrData).forEach(attrId => {
+                result[attrId] = {attributeId:attrId,counts: sampleAttrData[attrId], 'clinicalDataType': "SAMPLE"};
+            })
+            Object.keys(patientAttrData).forEach(attrId => {
+                result[attrId] = {attributeId:attrId,counts: patientAttrData[attrId], 'clinicalDataType': "PATIENT"};
```

Comment:
are you sure that a patient attr would never overwrite a sample attr with same name?